src/tests/test_utils/common_test_utils/CMakeLists.txt

```diff
@@ -39,6 +39,7 @@ function(add_common_utils ADD_TARGET_NAME)
                         "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
                     PRIVATE
                         "${CMAKE_CURRENT_SOURCE_DIR}/src"
+                        "${OpenVINO_SOURCE_DIR}/src/plugins/template/include"
```

Comment:
Use of `template` plugin makes dependency on `openvino_template_plugin` target - it should be added on CMake side
In this case `common_test_utils` becomes overcomplicated in terms of build. I suppose we need to move `graph_comparator` to a separate target, and make `infer_model_via_template` a part of `func_test_utils` (or `funcSharedTest`)

@ilya-lavrenov @iefode please, comment

Reply:
I agree that infer_model_via_template  should be in funcTestUtils not in the commonTestUtils.
But the problem is ov_nvidia_unit_tests uses this func. And I don't want to make ov_nvidia_unit_tests is dependent on funcTestUtils. Why unit test uses model inference - I don't know :)

Reply:
@vurusovs @olpipi 
1. It's a big mistake to use functional test utils in nvidia unit tests
2. Agree to move `openvino_template_plugin` dependency to `fun_test_utils` because temolate plugin could be used only in func tests
3. Look ok to move graph comparator to separated function
4. How to fix nvidia unit tests -> to move all  strange dependency directly to nvidia unit tests

Reply:
Please, add template plugin as a dependency, otherwise it won't be built from a CMake tree

Reply:
@olpipi Could you please check that we have removed template dependency from functional test utils cmake?